File: SimpleJavaConcurrent/src/main/java/com/example/simpleconcurrent/solutions/Worker.java
package com.example.simpleconcurrent.solutions;

import com.example.simpleconcurrent.utils.Console;

public class Worker extends Thread{
    private String mName;
    private Runnable mRunnable;
    public Worker(String name, Runnable runnable){
        this.mName = name;
        this.mRunnable = runnable;
    }
    public void run(){
        Console.log(mName + " start");
        mRunnable.run();
    }
}

File: SimpleKotlinConcurrent/src/main/java/com/example/simplekotlinconcurrent/solutions/Worker.kt
package com.example.simplekotlinconcurrent.solutions

import com.example.simplekotlinconcurrent.utils.Console

class Worker(name:String, runnable: Runnable):Thread(){
    private val mName:String = name;
    private val mRunnable:Runnable = runnable;

    override fun run() {
        Console.log("Worker " + mName + " started");
        mRunnable.run()
    }

}